internal/starter/loadConfig.go

```diff
@@ -0,0 +1,220 @@
+		i, _ := strconv.ParseUint(valueStr, 10, 0)
+		*p = uint(i)
+		return value
+	case *uint8:
+		i, _ := strconv.ParseUint(valueStr, 10, 8)
+		*p = uint8(i)
+		return value
+	case *uint16:
+		i, _ := strconv.ParseUint(valueStr, 10, 16)
+		*p = uint16(i)
+		return value
+	case *uint32:
+		i, _ := strconv.ParseUint(valueStr, 10, 32)
+		*p = uint32(i)
+		return value
+	case *uint64:
+		i, _ := strconv.ParseUint(valueStr, 10, 64)
+		*p = i
+		return value
+	case *float32:
+		i, _ := strconv.ParseFloat(valueStr, 32)
+		*p = float32(i)
+		return value
+	case *float64:
+		i, _ := strconv.ParseFloat(valueStr, 64)
+		*p = i
+		return value
+	case *string:
+		*p = valueStr
+		return value
+	default:
+		return any(valueStr).(T)
+	}
+}
+
+func GetEnv[T EnvTypes](key string, fallback T) T {
+	envValue := os.Getenv(key)
+	if envValue == "" {
+		return fallback
+	}
+	var value T
+	switch p := any(&value).(type) {
+	case *int:
+		i, _ := strconv.ParseInt(envValue, 10, 0)
+		*p = int(i)
+		return value
+	case *int8:
+		i, _ := strconv.ParseInt(envValue, 10, 8)
+		*p = int8(i)
+		return value
+	case *int16:
+		i, _ := strconv.ParseInt(envValue, 10, 16)
+		*p = int16(i)
+		return value
+	case *int32:
+		i, _ := strconv.ParseInt(envValue, 10, 32)
+		*p = int32(i)
+		return value
+	case *int64:
+		i, _ := strconv.ParseInt(envValue, 10, 64)
+		*p = i
+		return value
+	case *uint:
+		i, _ := strconv.ParseUint(envValue, 10, 0)
+		*p = uint(i)
+		return value
+	case *uint8:
+		i, _ := strconv.ParseUint(envValue, 10, 8)
+		*p = uint8(i)
+		return value
+	case *uint16:
+		i, _ := strconv.ParseUint(envValue, 10, 16)
+		*p = uint16(i)
+		return value
+	case *uint32:
+		i, _ := strconv.ParseUint(envValue, 10, 32)
+		*p = uint32(i)
+		return value
+	case *uint64:
+		i, _ := strconv.ParseUint(envValue, 10, 64)
+		*p = i
+		return value
+	case *float32:
+		i, _ := strconv.ParseFloat(envValue, 32)
+		*p = float32(i)
+		return value
+	case *float64:
+		i, _ := strconv.ParseFloat(envValue, 64)
+		*p = i
+		return value
+	case *string:
+		*p = envValue
+		return value
+	default:
+		return fallback
+	}
+}
+
+func LoadConfig(config interface{}) error {
+	valPtr := reflect.ValueOf(config)
+
+	if valPtr.Kind() != reflect.Ptr {
+		return ErrStructValPrt
+	}
+
+	val := valPtr.Elem()
+
+	if val.Kind() != reflect.Struct {
+		return ErrStructValPrt
+	}
+	typ := val.Type()
+
+	errStr := ""
+	for i := 0; i < typ.NumField(); i++ {
+		envVar := typ.Field(i).Tag.Get("env")
+		defValStr := typ.Field(i).Tag.Get("default")
+		name := typ.Field(i).Name
+
+		switch val.FieldByName(name).Type() {
+		case reflect.TypeOf(time.Nanosecond):
+			setVal := GetEnv(envVar, ConvertStringToType[string](defValStr))
+			setValDur, err := time.ParseDuration(setVal)
+			if err != nil {
+				errStr = fmt.Sprintf("%sfield name : %s , error: %s ;", errStr, name, err.Error())
+			} else {
+				val.FieldByName(name).SetInt(int64(setValDur))
+			}
+		default:
+			switch typ.Field(i).Type.Kind() {
+
+			case reflect.Int8:
```

Comment:
Additionally to what I mentioned before, would maybe [Viper](https://github.com/spf13/viper) be useful here, or some other similar package?

I have a feeling that this might significantly simplify our code (although I am not sure).

Reply:
Or this eventually: https://github.com/caarlos0/env

Reply:
Viper  doesn't support to set default via struct tag field, 
And https://github.com/caarlos0/env can be used , it fits our  requirement, 
But I don't want to add external dependency , so based on the  caarlos0/env package concept , I have created  LoadConfig function , To make it light weight and  tailer fit for our requirements

Reply:
https://github.com/caarlos0/env is used to load environment to config struct